Makefile.base

```diff
@@ -68,7 +68,9 @@ $(BINDIR)/$(MODULE)/:
 $(BINDIR)/$(MODULE).a $(OBJ): | $(BINDIR)/$(MODULE)/
 
 $(BINDIR)/$(MODULE).a: $(OBJ) | ${DIRS:%=ALL--%}
-	$(Q)$(AR) $(ARFLAGS) $@ $?
+	@# Recreate archive to cleanup deleted/non selected source files objects
+	$(Q)$(RM) $@
+	$(Q)$(AR) $(ARFLAGS) $@ $^
```

Comment:
Are we sure we want to remove it ALWAYS, and not only under certain conditions?

Reply:
Yes always, because we want to re-create a new archive. If not, old files are kept as `ar` is incremental.

Reply:
When doing `gcc -o out.o` it overwrites the previous object with the new one, and that is what we expect.
The RM here, is implementing an `--overwrite` option that `ar` does not have.